builtin/providers/spotinst/provider.go

```diff
@@ -0,0 +1,90 @@
+package spotinst
+
+import (
+	"os"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// Provider returns a terraform.ResourceProvider.
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"email": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: envDefaultFuncAllowMissing("SPOTINST_EMAIL"),
```

Comment:
There are actually functions built into core which we should leverage for this instead: https://github.com/hashicorp/terraform/pull/6098